dev.py

```diff
@@ -257,19 +257,7 @@ def main(argv):
         sys.exit(1)
 
     if args.doc:
-        try:
-            test, version, mod_path = runtests.import_module()
-        except ImportError:
-            # this may fail when running with --no-build, so try to detect
-            # an installed scipy in a subdir inside a repo
-            site_dir = get_site_packages()
-            print("Trying to build scipy docs from development installed "
-                  "path at:", site_dir)
-            sys.path.insert(0, site_dir)
-            os.environ['PYTHONPATH'] = \
-                os.pathsep.join((site_dir, os.environ.get('PYTHONPATH', '')))
-            test, version, mod_path = runtests.import_module()
-
+        test, version, mod_path = get_project_info()
```

Comment:
Instead of just fixing it for doc, I would call it after `if not args.no_build:`. This way we can import in both cases (build and no build). Or is there a specific reason you did not do that? (linked to your comment on no-build maybe? I did not fully get `this may fail when running with --no-build`)

Reply:
ok we are on the same page 👍